adoc/chapters/programming_interface.adoc

```diff
@@ -5186,718 +5173,637 @@ If no template argument is provided, then the default allocator for the SYCL
 
-a@
-[source]
+'''
+
+.[apititle]#buffer::reinterpret#
+[source,role=synopsis,id=api:buffer-reinterpret]
 ----
-template <typename ReinterpretT, int ReinterpretDim>
+template <typename ReinterpretT, int ReinterpretDim>                       (1)
 buffer<ReinterpretT, ReinterpretDim,
        typename std::allocator_traits<AllocatorT>::template rebind_alloc<
-           std::remove_const_t<ReinterpretT>>>
-reinterpret(range<ReinterpretDim> reinterpretRange) const
-----
-   a@ Creates and returns a reinterpreted SYCL [code]#buffer#
-      with the type specified by [code]#ReinterpretT#,
-      dimensions specified by [code]#ReinterpretDim# and range
-      specified by [code]#reinterpretRange#.  The buffer object
-      being reinterpreted can be a SYCL sub-buffer that was created
-      from a SYCL [code]#buffer# and must throw
-      [code]#exception# with the
-      [code]#errc::invalid# error code if the total
-      size in bytes represented by the type and range of the
-      reinterpreted SYCL [code]#buffer# (or sub-buffer) does not
-      equal the total size in bytes represented by the type and range
-      of this SYCL [code]#buffer# (or sub-buffer).
-      Reinterpreting a sub-buffer provides a reinterpreted view of
-      the sub-buffer only, and does not change the offset or size of
-      the sub-buffer view (in bytes) relative to the parent
-      [code]#buffer#.
-
-a@
-[source]
-----
-template <typename ReinterpretT, int ReinterpretDim = Dimensions>
+           ReinterpretT>>
+reinterpret(range<ReinterpretDim> reinterpretRange) const;
+
+template <typename ReinterpretT, int ReinterpretDim = Dimensions>          (2)
 buffer<ReinterpretT, ReinterpretDim,
        typename std::allocator_traits<AllocatorT>::template rebind_alloc<
-           std::remove_const_t<ReinterpretT>>>
-reinterpret() const
+           ReinterpretT>>
+reinterpret() const;
 ----
-   a@ Creates and returns a reinterpreted SYCL [code]#buffer#
-      with the type specified by [code]#ReinterpretT# and
-      dimensions specified by [code]#ReinterpretDim#.
-      Only valid when [code]#(ReinterpretDim == 1)# or when
-      [code]#\((ReinterpretDim == Dimensions) && (sizeof(ReinterpretT) == sizeof(T)))#.
-      The buffer object being reinterpreted can be a SYCL sub-buffer
-      that was created from a SYCL [code]#buffer#.  The implementation must
-      throw an [code]#exception# with the [code]#errc::invalid# error code
-      if the total size in bytes represented by this SYCL [code]#buffer# (or
-      sub-buffer) is not evenly divisible by [code]#sizeof(ReinterpretT)#.
-      Reinterpreting a sub-buffer provides a reinterpreted view
-      of the sub-buffer only,
-      and does not change the offset or size of the sub-buffer view (in bytes)
-      relative to the parent [code]#buffer#.
+_Preconditions (2):_ Available when [code]#(ReinterpretDim == 1)# or when
+[code]#\((ReinterpretDim == Dimensions) && (sizeof(ReinterpretT) ==
+sizeof(T)))#.
```

Comment:
Preconditions are for UB. Conditions that affect availability should be constraints.